src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcRecoveryPasswordDao.java

```diff
@@ -1,100 +1,62 @@
 package com.vinylteam.vinyl.dao.jdbc;
 
 import com.vinylteam.vinyl.dao.RecoveryPasswordDao;
-import com.vinylteam.vinyl.dao.jdbc.mapper.RecoveryTokenRowMapper;
+import com.vinylteam.vinyl.dao.jdbc.extractor.RecoveryTokenResultSetExtractor;
 import com.vinylteam.vinyl.entity.RecoveryToken;
-import com.zaxxer.hikari.HikariDataSource;
+import lombok.RequiredArgsConstructor;
 import lombok.extern.slf4j.Slf4j;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
 
 import java.sql.*;
 import java.time.Instant;
 import java.util.Optional;
 import java.util.UUID;
 
 @Slf4j
+@RequiredArgsConstructor
+@Repository
 public class JdbcRecoveryPasswordDao implements RecoveryPasswordDao {
 
     private static final String INSERT_TOKEN = "INSERT INTO recovery_password_tokens" +
             " (user_id, token, created_at)" +
-            " VALUES (?, ?, ?)" +
-            " ON CONFLICT (user_id) DO UPDATE SET token = ?, created_at = ?";
+            " VALUES (:user_id, :token, :created_at)" +
+            " ON CONFLICT (user_id) DO UPDATE SET token = :token, created_at = :created_at";
     private static final String FIND_BY_TOKEN = "SELECT id, user_id, token, created_at FROM recovery_password_tokens" +
-            " WHERE token = ?";
+            " WHERE token = :token";
     private static final String REMOVE_TOKEN = "DELETE FROM recovery_password_tokens" +
-            " WHERE id = ?";
+            " WHERE id = :id";
 
-    private static final RecoveryTokenRowMapper RECOVERY_ROW_MAPPER = new RecoveryTokenRowMapper();
-    private final HikariDataSource dataSource;
+    private static final RecoveryTokenResultSetExtractor RECOVERY_ROW_MAPPER = new RecoveryTokenResultSetExtractor();
```

Comment:
It is not RECOVERY_ROW_MAPPER , it is extractor. You should change name of value